sklearn/svm/src/libsvm/svm.h

```diff
@@ -145,6 +145,7 @@ void svm_set_print_string_function(void (*print_func)(const char *));
 
 /* sparse version */
 
+/* svm_csr_ functions are defined by libsvm_template.cpp from generic versions in svm.cpp */
```

Comment:
Same here

```suggestion
/* svm_csr_ functions are defined in libsvm_template.cpp with generic versions in svm.cpp */
```

Reply:
Aha. I thought "by from" avoid a confusion because the bodies are actually defined in svm.cpp, not in libsvm_template.cpp, who includes svm.cpp twice with different 'conditional flags'.

Let me know if you still prefer "in .. with".

Reply:
That makes sense. I'll merge your current version.